includes/class-wc-fgc-update-variation-cart.php

```diff
@@ -0,0 +1,387 @@
+		$wc_cart = WC()->cart->get_cart();
+
+		if ( ! empty( $wc_cart ) ) {
+			foreach ( $wc_cart as $cart_item_key => $cart_item ) {
+
+				// check that product is exists in the cart.
+				if ( $cart_item_key_ajax == $cart_item_key ) {
+					$_cart_item_key = $cart_item_key;
+					foreach ( $cart_item['variation'] as $key => $value ) {
+						$_REQUEST[ $key ] = $value;
+					}
+				}
+			}
+		}
+
+		// Get product and post from the id.
+		$product = wc_get_product( $product_id );
+
+		// Get post product from the id.
+		$post = get_post( $product_id );
+
+		/* setting global things on our own for getting things in woocommerce manner ::end */
+		do_action( 'wc_fgc_before_product_html' );
+		?>
+		<div class="wc_fgc_cart" data-title="<?php echo esc_attr( $product->add_to_cart_text() ); ?>" id="wc_fgc_<?php echo $_cart_item_key; ?>">
+			<section class="wc-fgc-close-section">
+				<a href="javascript:void(0)" class="wc-fgc-close-btn">
+					<span class="dashicons dashicons-after dashicons-no-alt"></span>
+				</a>
+			</section>
+			<div class="wc-fgc-stock-error" style="display: none;"></div>
+
+			<input type="hidden" id="wc_fgc_prevproid" value="<? echo $variation_id; ?>">
+			<input type="hidden" id="wc_fgc_cart_item_key" value="<?php echo $_cart_item_key; ?>">
+		<div itemscope itemtype="<?php echo woocommerce_get_product_schema(); ?>" id="product-<?php the_ID(); ?>" <?php post_class(); ?>>
+			<?php
+			/*
+			 * woocommerce_before_single_product_summary hook.
+			 * @hooked woocommerce_show_product_sale_flash - 10 , @hooked woocommerce_show_product_images - 20
+			 */
+			do_action( 'woocommerce_before_single_product_summary' );
+			?>
+			 <div class="summary entry-summary">
+				<?php
+				// get single product name.
+				wc_get_template( 'single-product/title.php' );
+				// get single product price.
+				wc_get_template( 'single-product/price.php' );
+
+				// update for selected attributes.
+				global $product;
+
+				// Enqueue variation scripts.
+				wp_enqueue_script( 'wc-add-to-cart-variation' );
+
+				// Get Available variations?
+				$get_variations = count( $product->get_children() ) <= apply_filters( 'woocommerce_ajax_variation_threshold', 30, $product );
+
+				// Load the template.
+				wc_get_template(
+					'single-product/add-to-cart/variable.php',
+					array(
+						'available_variations' => $get_variations ? $product->get_available_variations() : false,
+						'attributes'           => $product->get_variation_attributes(),
+					)
+				);
+				?>
+				 </div>
+				 <meta itemprop="url" content="<?php the_permalink(); ?>" />
+			 </div>
+			</div>
+			<?php
+
+			do_action( 'wc_fgc_after_product_html' );
+
+			wp_die();
+	}
+
+	/**
+	 * Handle ajax as per the updation in the cart.
+	 *
+	 * @name wc_fgc_update_variation_in_cart
+	 * @since 1.0.0
+	 */
+	public static function update_variation_in_cart() {
+		// verify the ajax request.
+		check_ajax_referer( 'wc-fgc-verify-nonce', 'nonce' );
+		global $woocommerce;
```

Comment:
what's this for? doesn't seem used. and if it is, should be `wc()`

Reply:
I think it was for updating the variation.